parser/ast/expressions.go

```diff
@@ -905,12 +905,16 @@ func (n *PatternLikeExpr) Restore(ctx *format.RestoreCtx) error {
 		return errors.Annotate(err, "An error occurred while restore PatternLikeExpr.Pattern")
 	}
 
-	escape := string(n.Escape)
-	if escape != "\\" {
-		ctx.WriteKeyWord(" ESCAPE ")
-		ctx.WriteString(escape)
-
+	escape := "?"
+	if _, ok := n.Escape.(ParamMarkerExpr); !ok {
+		// it can not convert to `ParamMarkerExpr`, so it must be `ValueExpr`
+		if n.Escape.(ValueExpr).GetString() == "\\" {
+			return nil
+		}
+		escape = n.Escape.(ValueExpr).GetString()
```

Comment:
Add the test case for testing `Restore`.

Reply:
Please check, THKS.